pkg/service/v0/service.go

```diff
@@ -260,19 +262,30 @@ func (p WopiServer) getWopiSrc(fileRef, viewMode, storageID, folderURL, userName
 
 	r, err := p.httpClient.Do(req)
 	if err != nil {
-		return "", err
+		return "", "", err
 	}
 	defer r.Body.Close()
 
 	if r.StatusCode != http.StatusOK {
-		return "", errors.New("get /wopi/iop/open failed: status code != 200")
+		return "", "", errors.New("get /wopi/iop/open failed: status code != 200")
 	}
 
 	b, err := ioutil.ReadAll(r.Body)
 	if err != nil {
-		return "", err
+		return "", "", err
+	}
+
+	path, err := url.PathUnescape(string(b))
+	if err != nil {
+		return "", "", errors.New("unescape failed")
 	}
-	return string(b), err
+
+	parts := strings.Split(path, "&access_token=")
```

Comment:
This is kind of hacky. ;)
It would be better to parse the key/value list into a map and take the value from that.

Reply:
path will be something like `https://foo.bar/foo/bar&access_token=eyabcdefg`. So there are no key-value pairs :-( Parsing with `url.Parse` does also not work, since the access_token is no valid query parameter

Reply:
You mean we receive a broken URL where the query parameters are appended with `&` instead of `?`?

Reply:
If so I get the hacky solution but then maybe we should create a fix for the broken URL.

Reply:
Its not a URL, it is just a string, intended to be appended to a URL-like string like that: `https://base.foo/baz?WOPISrc=`

As a result you will receive a valid URL with query params (but it still needs to be escaped obviously, did omit this for the example):
`https://base.foo/baz?WOPISrc=https://foo.bar/foo/bar&access_token=eyabcdefg`

Reply:
I could also first build the URL object and extract the access_token part from it. Would be more intuitive. I'll give it a try

Reply:
@C0rby The `strings.Split` logic is now gone. Instead `access_token` is removed from the final requests' query params.